forge-game/src/main/java/forge/game/card/CardProperty.java

```diff
@@ -178,6 +178,10 @@ public static boolean cardHasProperty(Card card, String property, Player sourceC
             if (!card.isRingBearer()) {
                 return false;
             }
+        } else if (property.equals("IsDoctor")) {
+            if (!card.getRules().isDoctor()) {
+                return false;
+            }
```

Comment:
Is this actually necessary? Harmonic Prodigy has "ValidCard$ Shaman.YouCtrl,Wizard.Other+YouCtrl", so wouldn't "ValidCard$ Doctor.YouCtrl" work fine?

Reply:
checking rules is very wrong, let me know if I should explain further